### PR TITLE
issues with privates

### PR DESCRIPTION
**Pull request by indeyets**

### CODE REVIEW

**Comment by indeyets** [comment]
test for #287 is here

**Comment by indeyets** [comment]
test for #288 is here too

**Comment by indeyets** [comment]
and, we're green

**Comment by clbn** [comment]
It would be really great to add some tests for ensuring that stuff DOES NOT sneak into timelines it shouldn't. Like, private posts to non-subscriber's homepage, alien posts to a specific-user timeline, etc.

**Comment by indeyets** [comment]
@clbn but we DO have such tests :) additional pull-requests are always welcome! ;)

**Comment by clbn** [comment]
@indeyets OK then!

File: app/models/post.js [hunk 1]
```diff
@@ -370,13 +378,19 @@ exports.addModel = function(database) {
...
+
+    // only subscribers are allowed to read direct posts
+    if (!await this.isStrictlyDirect()) {
+      let moreTimelineIds = await this.getCommentsFriendOfFriendTimelineIds(user)
+      timelineIds.push(...moreTimelineIds)
+
+      timelineIds = _.uniq(timelineIds)
     }
 
+    let timelines = await* timelineIds.map(id => models.Timeline.findById(id))
```
Comment (berkus) [inline app/models/post.js:392]: get rid of `await*`?

File: test/functional/mutual_friends.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ describe("MutualFriends", function() {
           })
       })
 
-      it('should not publish commented private post to likes feed', function(done) {
+      it('should not publish commented direct message to likes feed', function(done) {
```
Comment (berkus) [inline test/functional/mutual_friends.js]: likes feed -> comments feed

**Pull request merged**